### PR TITLE
Document hooks refactor

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR updates hooks and handlers documentation.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/158066) / small_tests / b01caf17e2dbd1117db46ea262538d4b703b0a30
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3903/158066/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3903/158066/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-15_13.04.01/lib.mongooseim.logs/run.2022-12-15_13.04.02/suite.log.html)

File: doc/developers-guide/Hooks-and-handlers.md [hunk 1]
```diff
@@ -125,23 +125,25 @@ That's usually done in, respectively, `start/2` and `stop/1` functions.
...
 It is clearly visible that the handler `inspect_packet` is added to the `offline_message_hook`.
 
 `HostType` is the one for which the handler will be executed.
 In the case of statically defined domains, it is the same as the host, as configured in the [`general.hosts` section](../configuration/general.md#generalhosts).
 
-The handler itself is specified as a module-function pair;
-the arity of the function is not specified at the registration nor verified when calling the handler.
-When writing a handler, the arity can be checked in the `mongoose_hooks` module.
+The handler itself is specified as a fun expression;
+the arity of the function is always 3 - more about actual arguments in [`Writing handlers`](#writing-handlers) section.
```
Comment (NelsonVides) [inline doc/developers-guide/Hooks-and-handlers.md:141]:
```suggestion
the arity of the function is always 3 - more about actual arguments in the [`Writing handlers`](#writing-handlers) section.
```

File: doc/developers-guide/Hooks-and-handlers.md [hunk 2]
```diff
@@ -170,47 +172,43 @@ Such skipped hooks update metrics are defined in the `mongoose_metrics_hooks` mo
...
     end.
 ```
 
-As seen in this example, a handler receives an accumulator, puts some value into it and returns it
-for further processing.
+As seen in this example, a handler receives an accumulator, parameters and extra parameters (in this case - ignored).
+Then it matches to the result of `process_subscription/4` and can return 3 different values:
+* `{ok, Acc}` - it allows further processing and does not change the accumulator.
+* `{stop, mongoose_acc:set(hook, result, false, Acc)}` - it stops further processing and returns accumulator with a new value in it.
+* `{stop, Acc}` - it stops furher processing and does not change the accumulator.
```
Comment (NelsonVides) [inline doc/developers-guide/Hooks-and-handlers.md:204]:
```suggestion
* `{stop, Acc}` - it stops further processing and does not change the accumulator.
```

**Review by chrzaszcz — COMMENTED** [review]
It looks good to me. I agree with what @NelsonVides suggested, and the branch needs rebasing because of conflicts.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/158215) / small_tests / b00b89606e9420974db6a9a3c396545b036793d4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3903/158215/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3903/158215/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-16_07.58.08/lib.mongooseim.logs/run.2022-12-16_07.58.08/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/158217) / small_tests / 0e97e00b927758cdad7e5c5c6307677697d4b7ac
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3903/158217/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3903/158217/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-12-16_07.59.22/lib.mongooseim.logs/run.2022-12-16_07.59.23/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**